halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/services/v1/ProviderService.java

```diff
@@ -42,21 +48,32 @@ public Provider getProvider(String deploymentName, String providerName) {
       provider = providers.getDockerRegistry();
     } else if (providerName.toLowerCase().equals("google")) {
       provider = providers.getGoogle();
+    } else {
```

Comment:
I dont like this enumeration here. I let it go in node/Providers.java. Could you pick one or the other so there is only one place to modify when adding a new provider?

For example here you could
provider = providers.find(providerName)
And add the above into the node/Providers. Then this class doesnt care who the providers are and doesnt need to be modified when adding a new provider.